xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -2400,23 +2399,115 @@ where
         action: SendWelcomesAction,
         message_cursor: Option<i64>,
     ) -> Result<(), GroupError> {
-        let welcomes = action
+        // Only encode welcome metadata once
+        let welcome_metadata = WelcomeMetadata {
+            message_cursor: message_cursor.unwrap_or(0) as u64,
+        };
+        let welcome_metadata_bytes = welcome_metadata.encode_to_vec();
+
+        let wp_capable = action
             .installations
-            .into_iter()
-            .map(
-                |installation| -> Result<WelcomeMessageInput, WrapWelcomeError> {
+            .iter()
+            .filter(|installation| {
+                installation
+                    .welcome_pointee_encryption_aead_types
+                    .compatible()
+            })
+            .count();
+
+        let (welcome_pointer_bytes, welcome_pointee) = if wp_capable
+            > xmtp_configuration::INSTALLATION_THRESHOLD_FOR_WELCOME_POINTER_SENDING
```

Comment:
now that the extractors are included -- do welcome pointers work with d14n or do we need to think about a switch on Client here?